naga/src/lib.rs

```diff
@@ -283,9 +283,15 @@ pub const BOOL_WIDTH: Bytes = 1;
 pub const ABSTRACT_WIDTH: Bytes = 8;
 
 /// Hash map that is faster but not resilient to DoS attacks.
-pub type FastHashMap<K, T> = rustc_hash::FxHashMap<K, T>;
+/// (Similar to rustc_hash::FxHashMap but using hashbrown::HashMap instead of std::collections::HashMap.)
+/// To construct a new instance: `FastHashMap::default()`
+pub type FastHashMap<K, T> =
+    hashbrown::HashMap<K, T, std::hash::BuildHasherDefault<rustc_hash::FxHasher>>;
+
 /// Hash set that is faster but not resilient to DoS attacks.
-pub type FastHashSet<K> = rustc_hash::FxHashSet<K>;
+/// (Similar to rustc_hash::FxHashSet but using hashbrown::HashSet instead of std::collections::HashMap.)
+pub type FastHashSet<K> =
+    hashbrown::HashSet<K, std::hash::BuildHasherDefault<rustc_hash::FxHasher>>;
```

Comment:
I noticed that egui is using hashers from `ahash` & `nohash-hasher`, started wondering if either of these would be worth considering. I also noticed what looks like some more hashers listed in: https://github.com/RustCrypto/hashes

I am guessing we should defer consideration to the future, just wanted to note this before I forget.

Reply:
Yeah probably. Using a special hasher likely needs to be supported by benchmarks anyway. I would be interested in seeing this investigated at some point. Could you make an issue?